docs/docs/dev/policies/traffic-metrics.md

```diff
@@ -332,6 +337,109 @@ Use the Discovery Service of [your choice](https://prometheus.io/docs/prometheus
+kind: Mesh
+metadata:
+  name: default
+spec:
+  metrics:
+    enabledBackend: prometheus-1
+    backends:
+    - name: prometheus-1
+      type: prometheus
+      conf:
+        skipMTLS: false
+        port: 5670
+        path: /metrics
+        tags: # tags that can be referred in Traffic Permission when metrics are secured by mTLS 
+          kuma.io/service: dataplane-metrics
+        aggregate:
+          my-service: # name of the metric, required to later disable/override at dataplane configuration
+            path: "/metrics/prometheus"
+            port: 8888
+          other-sidecar:
+            # default path is going to be used, default: /metrics
+            port: 8000
+```
+:::
+::: tab "Universal"
+```yaml
+type: Mesh
+name: default
+metrics:
+  enabledBackend: prometheus-1
+  backends:
+  - name: prometheus-1
+    type: prometheus
+    conf:
+      port: 5670
+      path: /metrics
+      skipMTLS: true # by default mTLS metrics are also protected by mTLS. Scraping metrics with mTLS without transparent proxy is not supported at the moment.
+      aggregate:
+        my-service: # name of the metric, required to later disable/override at dataplane configuration
+          path: "/metrics/prometheus"
+          port: 8888
+        other-sidecar:
+          # default path is going to be used, default: /metrics
+          port: 8000
+```
+:::
+::::
+
+This configuration will cause every application in the `Mesh` to be scrapped for metrics by the Kuma dataplane. If you need to expose metrics only for the specific application it is possible through `annotation` for Kubernetes or `Dataplane` configuration for Universal deployment.
+
+:::: tabs :options="{ useUrlFragment: false }"
+::: tab "Kubernetes"
+Kubernetes allows to configure it through annotations. In case to configure you can use `prometheus.metrics.kuma.io/aggregate-<name>-(path/port/enabled)`, where name is used to match the `Mesh` configuration and override or disable it.
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+ namespace: kuma-example
+ name: kuma-tcp-echo
+spec:
+ ...
+ template:
+   metadata:
+     ...
+     annotations:
+       prometheus.metrics.kuma.io/aggregate-my-service-enabled: "false"  # causes that configuration from Mesh is disabled and endpoint's metrics won't be exposed
+       prometheus.metrics.kuma.io/aggregate-other-sidecar-port: "1234" # override port from Mesh
+       prometheus.metrics.kuma.io/aggregate-application-port: "80"
+       prometheus.metrics.kuma.io/aggregate-application-path: "/stats"
+   spec:
+     containers:
+     ...
+```
+:::
+::: tab "Universal"
+```yaml
+type: Dataplane
+mesh: default
+name: example
+metrics:
+ type: prometheus
+ conf:
+   path: /metrics/overridden
+   aggregate:
+     my-service: # causes that configuration from Mesh is disabled and endpoint's metrics won't be exposed
+       enabled: false
+     other-sidecar:
+       port: 1234 # override port from Mesh
+     application:
```

Comment:
We screwed up a little, we should have made this an array, that's the recommended way in CRDs... Array should be:

```yaml
- name: my-service
   enabled: false
   port: 1234
 ```

Reply:
My bad didn't know about recommended way. Should I create an issue that we should in the future change this to the list as a no-breaking change(new field)?